test/NuGet.Clients.Tests/NuGet.CommandLine.Test/RestoreNETCoreTest.cs

```diff
@@ -8958,6 +8958,78 @@ public void RestoreNetCore_PackagesLockFile_WithProjectChangeFramweorksAndLocked
             }
         }
 
+
+
+        [Fact]
+        public async void RestoreNetCore_PackagesLockFile_WithReorderedRuntimesInLockFile_PassRestore() { 
+            // A project with RestoreLockedMode should pass restore if the runtimes in the lock file have been reordered
+            using (var pathContext = new SimpleTestPathContext())
+            {
+
+                // Set up solution, project, and packages
+                var solution = new SimpleTestSolutionContext(pathContext.SolutionRoot);
+                var projFramework = Frameworks.FrameworkConstants.CommonFrameworks.Net461.GetShortFolderName();
+
+                var projectA = SimpleTestProjectContext.CreateNETCore(
+                   "a",
+                   pathContext.SolutionRoot,
+                   NuGetFramework.Parse(projFramework));
+
+                projectA.Properties.Add("RestorePackagesWithLockFile", "true");
+
+                // Set up the package and source
+                var packageX = new SimpleTestPackageContext()
+                {
+                    Id = "x",
+                    Version = "1.0.0"
+                };
+                packageX.Files.Clear();
+                packageX.AddFile("lib/netcoreapp2.0/x.dll");
+                packageX.AddFile("ref/netcoreapp2.0/x.dll");
+                packageX.AddFile("lib/net461/x.dll");
+                packageX.AddFile("ref/net461/x.dll");
+
+                await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+                    pathContext.PackageSource,
+                    PackageSaveMode.Defaultv3,
+                    packageX);
+
+                projectA.AddPackageToAllFrameworks(packageX);
+
+
+                solution.Projects.Add(projectA);
+                solution.Create(pathContext.SolutionRoot);
+
+                // Act
+                var r = Util.RestoreSolution(pathContext);
+
+
+                Assert.True(File.Exists(projectA.NuGetLockFileOutputPath));
+                var packagesLockFile = PackagesLockFileFormat.Read(projectA.NuGetLockFileOutputPath);
+
+                //Modify the list of target/runtimes so they are reordered in the lock file.
+                Assert.True(packagesLockFile.Targets.Count >=3); //Since we are removing the second one and adding to the end.
```

Comment:
I think if you provide the runtimes explicitly and in a revert alphabetical order you hit this issue out of the box :)

Reply:
See changes.  I had to keep the Null RID on top.  If it wasn't it would fail [here](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.ProjectModel/ProjectLockFile/PackagesLockFileUtilities.cs#L97) because the Target Framework is associated with all RIDS when the lock file is created.  So this [line](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.ProjectModel/ProjectLockFile/PackagesLockFileUtilities.cs#L86) was matching on a different RID I believe, and I didnt want to include that in here.  Not sure if that is another bug.